core/modules/qmeta/QMeta.h

```diff
@@ -156,17 +160,53 @@ class QMeta  {
      *
      *  @param queryId:   Query ID, non-negative number.
      */
-    virtual void finishQuery(int queryId) = 0;
+    virtual void finishQuery(QueryId queryId) = 0;
 
     /**
-     *  @brief Find all queries currently executing.
+     *  @brief Generic interface for finding queries.
+     *
+     *  Returns set of query IDs which satisfy all selections specified in parameters.
+     *
+     *  Setting "completed" to true is equivalent to setting "status" to a set of
+     *  (COMPLETED, FAILED, ABORTED) but is based on different QInfo attribute,
+     *  it uses "completed" instead of "status". Similarly setting "completed"
+     *  to false is equivalent to setting "status" to (EXECUTING).
```

Comment:
I find this to be confusing. "completed" and "returned" are not booleans, and yet you describe them as such. Why not go with a vector of things to match like "status", or bitfields for all 3 parameters?